view/loadView.js

```diff
@@ -108,6 +110,12 @@ function samlAuthentication(userProfile, done) {
       });
     }
 
+    if (!user.fullName) {
+      user.update({
```

Comment:
The "done" on line 120 is going to be called *before* the user.update call executes unless you use promise semantics i.e. update(...).then(...)